src/user/user.service.ts

```diff
@@ -132,17 +172,30 @@ export class UserService {
         data: data,
       });
     } catch (error) {
-      throw new HttpException('Failed to update user.', HttpStatus.INTERNAL_SERVER_ERROR);
+      if (error instanceof NotFoundException || error instanceof UnauthorizedException) throw error;
+      throw new GlobalException(ErrorsTypes.USER_PROFILE_FAILED_TO_UPDATE);
     }
   }
 
-  async removeUserProfile(id: string) {
+  async removeUserProfile(id: string, user: User) {
     try {
-      return await this.prismaService.userProfile.delete({
+      console.log('user', user);
+      if (!user) {
```

Comment:
the same as for update. not found issue we can throw if information wasn't find in db, not related to producer authorization

Reply:
fixed, now admin can update/delete anyone's profile and user's can only update/delete their own profile also changed to forbidden exception